File: frontend/lib/models/service.dart
class Service {
  final String id;
  final String name;
  final String category;
  final String description;
  final double price;
  final String timeline;
  final List<String> useCases;
  final int requestCount;
  final DateTime createdAt;

  Service({
    required this.id,
    required this.name,
    required this.category,
    required this.description,
    required this.price,
    required this.timeline,
    this.useCases = const [],
    this.requestCount = 0,
    required this.createdAt,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['_id'],
      name: json['name'],
      category: json['category'],
      description: json['description'],
      price: (json['price'] as num).toDouble(),
      timeline: json['timeline'],
      useCases: List<String>.from(json['useCases'] ?? []),
      requestCount: json['requestCount'] ?? 0,
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'category': category,
      'description': description,
      'price': price,
      'timeline': timeline,
      'useCases': useCases,
      'requestCount': requestCount,
    };
  }
}

File: backend/models/Service.js
const mongoose = require('mongoose');

const serviceSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  category: { type: String, enum: ['ai_dev', 'digital_marketing'], required: true },
  description: { type: String, required: true },
  price: { type: Number, required: true },
  timeline: { type: String, required: true },
  useCases: [{ type: String }],
  requestCount: { type: Number, default: 0 },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Service', serviceSchema);
